WrightTools/data.py

```
@@ -2861,9 +2861,79 @@ def from_shimadzu(filepath, name=None, verbose=True):
+        Default is None.
+    verbose : boolean (optional)
+        Toggle talkback. Default is True.
+
+    Returns
+    -------
+    WrightTools.data.Data object
+    """
+    # check filepath
+    if not os.path.isfile(filepath):
+        raise wt_exceptions.FileNotFound(path=filepath)
+    if not filepath.endswith('asc'):
+        wt_exceptions.WrongFileTypeWarning.warn(filepath, 'asc')
+    # set unspecified delimiter parameter arg to ','
+    if delimiter is None:
+        delimiter = ','
+    else:
+        pass
+    # import data
+    # now import file as a local var as comma-delimited .asc file
+    arr = np.genfromtxt(filepath,
+                        skip_header=10, skip_footer=1,
+                        delimiter=delimiter).T
+    # unexpected delimiter handler
+    if np.any(np.isnan(arr)):
+        # delimiter warning dictionary
+        delim_args = [',', '', '\t', ';', ':']
+        delim_strs = ['comma', 'space', 'tab', 'semicolon', 'colon']
+        delim_dict = dict(zip(delim_args, delim_strs))
+        if verbose:
+            print("Error: file is not %s-delimited!\n"
+                  "Trying other delimiters "
+                  "in wt.data.from_spc130() call." % delim_dict[delimiter])
+        for delimiter in delim_args:
+            arr = np.genfromtxt(filepath,
+                                skip_header=10, skip_footer=1,
+                                delimiter=delimiter).T
+            if not np.any(np.isnan(arr)):
+                if verbose:
+                    print("Error resolved: file is %s-delimited."
+                          % delim_dict[delimiter])
+                break
+    if np.any(np.isnan(arr)):
+        print("Error unresolved: Please check that your file "
+              "is formatted properly. An example of SPC-130 file format "
+              "can be found in ..\WrightTools\WrightTools\datasets\spc130"
+              "\n")
+        print('data object not created!\n')
```

you should raise an exception here---the user should not be allowed to proceed

`raise RuntimeError('your message here')

---

Yes, then remove the else block, there should be at least one return that is not in a conditional block, usually (some old-school programmers will say that there should be exactly one return statement for a given method, but I think that harms more often than it helps readability)